### DISCUSSION TITLE
Make Matplotlib Output a Figure

### DISCUSSION THREAD

**Discussion by lu-kas** [executablebooks/meta#225] [Q&A]
Dear all,

I would like to be able to add a caption and label (for later reference) to a graphics created by matplotlib. So far, the way I see to do that is to create the matplotlib figure, hide the cell output and then use glue to create a figure. 

Does anyone know a more convenient way to do this? I.e. is it possible to create a referencable figure directly from the matplotlib output?

Thanks!

**Comment by chrisjsewell** [comment]
Yes you should be able to use https://myst-nb.readthedocs.io/en/latest/use/formatting_outputs.html#images

**Reply by lu-kas** [reply]
Thanks for your quick response. I have adjusted the cell metadata and got now a caption. Yet, what I did not explicitly mention in the original question, I would like to have a figure number added. Do you know if and how this can be achieved?
